pkg/kube/init.go

```diff
@@ -25,16 +25,12 @@ import (
 
 // KubeClient contains some objects used to communicate with Kubernetes API Server.
 type KubeClient struct {
+	cfg *config.Config
+
 	// Client is the object used to operate Kubernetes builtin resources.
 	Client kubernetes.Interface
 	// APISIXClient is the object used to operate resources under apisix.apache.org group.
 	APISIXClient clientset.Interface
-	// SharedIndexInformerFactory is the index informer factory object used to watch and
```

Comment:
Why change this?

Reply:
```
// in controller
// re-execute this code will make panic `close of closed channel` when context cancelled
// unless you re-create Informer
**Informer.Run(ctx.Done())
```

So should re-create **Informer in every controller.run. It need re-create InformerFactory.

Reply:
OK, got it.